Utils/Dataflow/091_datasetsRucio/datasets_processing.py

```diff
@@ -140,15 +142,33 @@ def process_output_ds(stage, message):
     if type(datasets) != list:
         datasets = [datasets]
 
-    for dataset in datasets:
-        ds = get_output_ds_info(dataset)
+    mfields = META_FIELDS[OUTPUT]
+    for ds_name in datasets:
+        incompl = None
+        try:
+            ds = get_ds_info(ds_name, mfields)
+        except RucioException, err:
+            stage.log(["Mark message as incomplete (failed to get information"
+                       " from Rucio for: %s)." % ds_name,
+                       "Reason: %s." % str(err)],
```

Comment:
Minor thing: wouldn't it be better to change the order of two parts of the first item? Form "Failed to do X, doing Y" seems more natural to me than "Doing Y (because X occurred)" in case of a continuous log, like here.

Reply:
@Evildoor,
Maybe you're right from the "natural" language point of view, but this is about log records.
There may be different reasons to mark message as incomplete; and the more "general" message looks better (for me) at the beginning, not at the end.

Just like (unhandler) exception output:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: Some explaination here.
```
-- first you see that it is an `IndexError`, and (unless someone uses random exceptions) already get the general idea: it's gonna be something about attempt to use list element that does not exists (maybe empty list?). But if first you see something like "Function XXX returned empty list, which we didn't expect, so here we are: IndexError." -- you have no clue to what happened unless the message is fully read and understood. The faster you see keywords that tell you what's the whole situation is about -- the better; and here the situation is about "The message will be passed with 'incomplete' marker". If you need to know more -- you'll read further. If you know that "incomplete" messages has nothing to do with you current issue and/or are automatically fixed after a while -- read first words and skip the rest.

Does it sound reasonable or am I just trying to excuse myself? ;)

Reply:
Fair enough.